sktime/forecasting/naive.py

```diff
@@ -139,5 +143,40 @@ def _predict_last_window(self, fh, X=None, return_pred_int=False,
             fh_idx = fh.index_like(self.cutoff)
             return last_window[fh_idx]
 
-        elif self.strategy == "mean":
+        elif self.strategy == "mean" and self.sp == 1:
             return np.repeat(np.nanmean(last_window), len(fh))
+
+        elif self.strategy == "mean" and self.sp != 1:
+            last_window = last_window.astype(float)
+            last_window = np.pad(last_window,
```

Comment:
Why do we pad with -1? I think I prefer the solution where we raise an error when window length is not a multiple of sp, seems more intuitive and easier in terms of the code, but of course feel free to disagree

Reply:
we can pad it with any value, perhaps we can use categorical value also . its purpose is for distinguishing the padded elements as to make it easier to identify them while imputing values into them such that the mean with imputed padded elements returns the mean which is expected to be. for example:
let the last window be [3,4,5,6,7,8,9] and sp be 3,
now if we try to reshape this last window into 3*3 matrix it cant be done, hence we pad elements
we will pad with any negative number because time series data is seldom negative
last window on padding say -1 [3,4,5,6,7,8,9,-1,-1] makes this 3*3=9 which fits
now lets reshape this
last window is 3,4,5
                       6,7,8
                       9,-1,-1
for making the mean robust we will impute elements into places where -1 occurs such that mean remains the same
i.e consider mean of column 4,7,-1 expected mean is 4+7 /2=5.5 we impute -1 with a value 'x' computed such that mean  of 4,7,x is 5.5 finding x gives 5.5 and replace it in place of -1. 
now the column is 4,7,5.5  ,notice that if it wasn't for float the imputed value would've been just 5 and not 5.5
now lastwindow.mean(axis=0) gives the right values .
p.s: this is just for 3*3 reshaped matrix, it is different for 4*4 i.e if sp is 4 and so on
i've modelled this equation by considering mathematical induction.
Hence it works for all other sp values too

Reply:
I think this implementation considers the case where window length is a multiple of sp and also the other case where the user wants to seasonal mean with window length not being a sp multiple. And such small differences are likely to favour more usage of our tool.

Reply:
Maybe documentation could be made self explanantory.